core/src/main/java/com/linecorp/armeria/client/retry/RetryRule.java

```diff
@@ -0,0 +1,215 @@
+    }
+
+    /**
+     * Returns a newly created {@link RetryRule} that will retry with the
+     * {@linkplain Backoff#ofDefault() default backoff} if the response status is one of
+     * the specified {@link HttpStatus}es.
+     */
+    static RetryRule onStatus(Iterable<HttpStatus> statuses) {
+        return builder().onStatus(statuses).thenBackoff();
+    }
+
+    /**
+     * Returns a newly created a {@link RetryRule} that will retry with the
+     * {@linkplain Backoff#ofDefault() default backoff} if the response status matches
+     * the specified {@code statusFilter}.
+     */
+    static RetryRule onStatus(Predicate<? super HttpStatus> statusFilter) {
+        return builder().onStatus(statusFilter).thenBackoff();
+    }
+
+    /**
+     * Returns a newly created a {@link RetryRule} that will retry with
+     * the {@linkplain Backoff#ofDefault() default backoff} if an {@link Exception} is raised and
+     * that is instance of the specified {@code exception}.
+     */
+    static RetryRule onException(Class<? extends Throwable> exception) {
+        return builder().onException(exception).thenBackoff();
+    }
+
+    /**
+     * Returns a newly created {@link RetryRule} that will retry with the
+     * {@linkplain Backoff#ofDefault() default backoff} if an {@link Exception} is raised and
+     * the specified {@code exceptionFilter} returns {@code true}.
+     */
+    static RetryRule onException(Predicate<? super Throwable> exceptionFilter) {
+        return builder().onException(exceptionFilter).thenBackoff();
+    }
+
+    /**
+     * Returns a newly created {@link RetryRule} that retries with
+     * {@linkplain Backoff#ofDefault() default backoff} on any {@link Exception}.
+     * Note that this rule should be used carefully because it reties regardless of
+     * <a href="https://developer.mozilla.org/en-US/docs/Glossary/Idempotent">idempotency</a>.
+     */
+    static RetryRule onException() {
+        return builder().onException().thenBackoff();
+    }
+
+    /**
+     * Returns a {@link RetryRule} that retries with the {@linkplain Backoff#ofDefault() default backoff}
+     * on an {@link UnprocessedRequestException} which means that the request has not been processed by
+     * the server. Therefore, you can safely retry the request without worrying about the idempotency of
+     * the request.
+     */
+    static RetryRule onUnprocessed() {
+        return builder().onUnprocessed().thenBackoff();
+    }
+
+    /**
+     * Returns a newly created {@link RetryRuleBuilder}.
+     */
+    static RetryRuleBuilder builder() {
+        return new RetryRuleBuilder();
+    }
+
+    /**
+     * Returns a composed {@link RetryRule} that represents a logical OR of this {@link RetryRule} and another.
+     * If this {@link RetryRule} completes with {@link RetryRuleDecision#next()}, then other {@link RetryRule}
+     * is evaluated.
+     */
+    default RetryRule or(RetryRule other) {
```

Comment:
Could we also add `static RetryRule of(RetryRule...)` and `static RetryRule of(Iterable<? extneds RetryRule>)`, which yield prettier indentation?

Reply:
SGTM!